src/Symfony/Component/DependencyInjection/Compiler/AbstractRecursivePass.php

```diff
@@ -112,6 +121,10 @@ protected function getConstructor(Definition $definition, bool $required): ?\Ref
         }
 
         if (\is_string($factory = $definition->getFactory())) {
+            if (str_starts_with($factory, '@=')) {
+                return new \ReflectionFunction(static function () {});
```

Comment:
This should probably be updated since this PR support passing arguments to factory:
```suggestion
                return new \ReflectionFunction(static function (...$args) {});
```

Reply:
Why not. Updated.